### DISCUSSION TITLE
.net 8 and the new windows RID

### DISCUSSION THREAD

**Discussion by brabebhin** [ffmpeginteropx/FFmpegInteropX#396] [General]
So in .net 8, there's a breaking change which removes some platform RIDs.
https://learn.microsoft.com/en-gb/dotnet/core/rid-catalog#windows-rids

Basically, the win10-{platform} RID no longer exists.
I have a dark feeling this breaks both our nuget package and the direct reference trickery. 
It causes REG_DB_CLASSNOTREG exceptions.

I think we need a different package for .net 8 onwards.
Unfortunately, .net 6 is still supported until late 2024, so we can't just drop support (or we can? since we never really supported it officially?)

**Comment by lukasf** [comment]
From what I have read, it seems that .net 6 has already supported the "portable RID" ("win" instead of "win10"). I hope it will be sufficient to switch to "win" RID and things should work on .net 6 and upwards. But this needs to be seen.

**Comment by lukasf** [comment]
D'oh, looks like even Microsoft's own WindowsAppSdk ist not even updated for .net 8, same issue about RID when used in .net 8 projects. I think we should wait and see how this is done in the next WindowsAppSdk release, and take over that solution.

**Reply by brabebhin** [reply]
Yeah, i noticed the fix isn't quite straightforward, having wasted several hours on this.
